eZ/Publish/Core/Persistence/Legacy/User/Handler.php

```diff
@@ -253,29 +300,86 @@ public function loadRoles()
     }
 
     /**
-     * Update role.
+     * Update role (draft).
      *
      * @param \eZ\Publish\SPI\Persistence\User\RoleUpdateStruct $role
+     * @param int $status One of Role::STATUS_DEFINED|Role::STATUS_DRAFT
      */
-    public function updateRole(RoleUpdateStruct $role)
+    public function updateRole(RoleUpdateStruct $role, $status = Role::STATUS_DEFINED)
     {
-        $this->roleGateway->updateRole($role);
+        $this->roleGateway->updateRole($role, $status);
     }
 
     /**
-     * Delete the specified role.
+     * Delete the specified role (draft).
      *
      * @param mixed $roleId
+     * @param int $status One of Role::STATUS_DEFINED|Role::STATUS_DRAFT
      */
-    public function deleteRole($roleId)
+    public function deleteRole($roleId, $status = Role::STATUS_DEFINED)
```

Comment:
This should remove policies for the given version only, `$status` is not passed to the `Gateway::removePolicy()`.
The case should be tested as well.